src/gui/labeling/qgsrulebasedlabelingwidget.cpp

```diff
@@ -461,7 +480,12 @@ bool QgsRuleBasedLabelingModel::setData( const QModelIndex &index, const QVarian
 
   if ( role == Qt::CheckStateRole )
   {
-    rule->setActive( value.toInt() == Qt::Checked );
+    if ( QGuiApplication::keyboardModifiers() == Qt::ShiftModifier || QGuiApplication::keyboardModifiers() == Qt::ControlModifier )
```

Comment:
I find this logic a little strange -- we don't use this approach anywhere else. Couldn't we just go with the "space=toggle all selected" approach we use for the layer tree here too?